makefile

```diff
@@ -0,0 +1,195 @@
+USER_SOURCES := $(shell find ./$(DEVICE_DIRNAME)/Core/Src -name "*.c")
+
+HAL_SOURCES := $(shell find ./$(DEVICE_DIRNAME)/Drivers/STM32F0xx_HAL_Driver/Src -name "*.c")
+
+C_SOURCES += $(HAL_SOURCES)
+C_SOURCES += $(USER_SOURCES)
+
+# ASM sources
+ASM_SOURCES = $(DEVICE_DIRNAME)/Core/Startup/startup_stm32f042k4tx.s
+
+
+#######################################
+# binaries
+#######################################
+PREFIX = arm-none-eabi-
+# The gcc compiler bin path can be either defined in make command via GCC_PATH variable (> make GCC_PATH=xxx)
+# either it can be added to the PATH environment variable.
+ifdef GCC_PATH
+CC = $(GCC_PATH)/$(PREFIX)gcc
+AS = $(GCC_PATH)/$(PREFIX)gcc -x assembler-with-cpp
+CP = $(GCC_PATH)/$(PREFIX)objcopy
+SZ = $(GCC_PATH)/$(PREFIX)size
+else
+CC = $(PREFIX)gcc
+AS = $(PREFIX)gcc -x assembler-with-cpp
+CP = $(PREFIX)objcopy
+SZ = $(PREFIX)size
+endif
+HEX = $(CP) -O ihex
+BIN = $(CP) -O binary -S
+ 
+#######################################
+# CFLAGS
+#######################################
+# cpu
+CPU = -mcpu=cortex-m0
+
+# fpu
+#FPU = -mfpu=fpv4-sp-d16
+
+# float-abi
+FLOAT-ABI = -mfloat-abi=soft
+
+# mcu
+MCU = $(CPU) -mthumb $(FPU) $(FLOAT-ABI)
+
+# macros for gcc
+# AS defines
+AS_DEFS = 
+
+# C defines
+# C_DEFS = \
+# -D USE_HAL_DRIVER \
+# -D STM32F401xE
+
+C_DEFS = \
+-D USE_HAL_DRIVER \
+-D STM32F042x6 \
+-D DEBUG \
+-D POD_INDICATOR_LIGHT
+
+
+# AS includes
+AS_INCLUDES = 
+
+# C includes
+C_INCLUDES =  \
+-I ./$(DEVICE_DIRNAME)/Core/Inc \
+-I ./$(DEVICE_DIRNAME)/Drivers/STM32F0xx_HAL_Driver/Inc \
+-I ./$(DEVICE_DIRNAME)/Drivers/STM32F0xx_HAL_Driver/Inc/Legacy \
+-I ./$(DEVICE_DIRNAME)/Drivers/CMSIS/Device/ST/STM32F0xx/Include \
+-I ./$(DEVICE_DIRNAME)/Drivers/CMSIS/Include \
+-I ./WLoopCAN/include
+
+C_INCLUDES += $(USER_INCLUDES)
+
+# compile gcc flags
+ASFLAGS = $(MCU) $(AS_DEFS) $(AS_INCLUDES) $(OPT) -Wall -fdata-sections -ffunction-sections
+
+CFLAGS = $(MCU) $(C_DEFS) $(C_INCLUDES) $(OPT) -Wall -fdata-sections -ffunction-sections
+
+ifeq ($(DEBUG), 1)
+CFLAGS += -g -gdwarf-2
+endif
+
+
+# Generate dependency information
+CFLAGS += -MMD -MP -MF"$(@:%.o=%.d)"
+
+
+#######################################
+# LDFLAGS
+#######################################
+# link script
+LDSCRIPT = $(DEVICE_DIRNAME)/STM32F042K4TX_FLASH.ld
+
+# libraries
+LIBS = -lc -lm -lnosys 
+LIBDIR = 
+LDFLAGS = $(MCU) -specs=nano.specs -T$(LDSCRIPT) $(LIBDIR) $(LIBS) -Wl,-Map=$(BUILD_DIR)/$(TARGET).map,--cref -Wl,--gc-sections
+
+# default action: build all
+all: $(BUILD_DIR)/$(TARGET).elf $(BUILD_DIR)/$(TARGET).hex $(BUILD_DIR)/$(TARGET).bin
+
+
+#######################################
+# build the application
+#######################################
+# list of objects
+OBJECTS = $(addprefix $(BUILD_DIR)/,$(notdir $(C_SOURCES:.c=.o)))
+vpath %.c $(sort $(dir $(C_SOURCES)))
+# list of ASM program objects
+OBJECTS += $(addprefix $(BUILD_DIR)/,$(notdir $(ASM_SOURCES:.s=.o)))
+vpath %.s $(sort $(dir $(ASM_SOURCES)))
+
```

Comment:
Can you make it so that this makefile builds the submodule for you? You can define a target that's required by the linking step:

```makefile
can_library:
    cd WLoopCAN && make pod_indicator_lights
```

Then add it as a requirement for `$(BUILD_DIR)/$(TARGET).elf`

Reply:
- [ ] add WLoopCAN submodule make target into root make in makefile